rs/replicated_state/src/canister_state/execution_state.rs

```diff
@@ -477,6 +478,14 @@ pub struct ExecutionState {
 
     /// Round-robin across canister method types.
     pub next_scheduled_method: NextScheduledMethod,
+
+    /// Checks if execution is in Wasm64 mode. This field is used as a cache to avoid
+    /// parsing the wasm module to determine if it is a Wasm64 module.
+    /// Should not be accessed directly, use `is_wasm64()` method instead.
+    /// Also this field does not need to be saved in the checkpoint/state
+    /// as it can always be recomputed from the wasm module.
+    #[validate_eq(Ignore)]
+    pub is_wasm64: Option<bool>,
```

Comment:
How do you guarantee that this field doesn't become stale when e.g. upgrading or reinstalling a canister with a different execution mode? Wouldn't it be better if you put it into `WasmBinary` instead, as this definitely needs to be replaced when changing the binary?

Reply:
the function `create_execution_state()` is called whenever a canister is installed/upgraded/etc and here we set the field again. (I wanted to share a link to that change, but I don't know how, it's here: `rs/canister_sandbox/src/replica_controller/sandboxed_execution_controller.rs` )

Reply:
it's now saved in the state, I guess this doesn't apply anymore.